src/databricks/sql/cloudfetch/downloader.py

```diff
@@ -0,0 +1,103 @@
+import logging
+
+import requests
+import lz4.frame
+import threading
+import time
+
+logger = logging.getLogger(__name__)
+
+
+class ResultSetDownloadHandler(threading.Thread):
+
+    def __init__(self, downloadable_result_settings, t_spark_arrow_result_link):
+        super().__init__()
+        self.settings = downloadable_result_settings
+        self.result_link = t_spark_arrow_result_link
+        self.is_download_scheduled = False
+        self.is_download_finished = threading.Event()
+        self.is_file_downloaded_successfully = False
+        self.is_link_expired = False
+        self.is_download_timedout = False
+        self.http_code = None
```

Comment:
When we call the .run() method, these properties always set the same way. Do they need to be initialised here? Should they instead be initialised only in the run method? Or only in this method?

Reply:
I think they all need to be initialized because upstream, `ResultFileDownloadManager` may set some of these properties, and because of multithreading, there can be scenarios where `is_file_download_successful()` looks at properties that `run()` may not have set yet. Also, `run()` on a handler may be called multiple times depending on retries. I'll keep the initialization here but will create a `_reset()` function that resets some of these properties